src/web/routes/application/select-address/adapters.js

```diff
@@ -0,0 +1,75 @@
+const { props, propOr, pickAll, compose, map, join, filter, not, isEmpty, split, head, tail, defaultTo } = require('ramda')
+const { THOROUGHFARE_PROPS, BUILDING_NAME_PROPS, LOCALITY_PROPS, DELIMITER } = require('./constants')
+const { OS_PLACES_ADDRESS_KEYS } = require('../common/constants')
+
+/**
+ * Map address returned from OS places API to format required by claimant service
+ * N.B. assumes address has been normalised by /postcode adapter
+ *
+ * 1. Normalise address by setting any missing fields to a default value of an empty string
+ * 2. Construct the 4 possible address parts for address lines 1 and 2:
+ *    a) Organisation name
+ *    b) Building name
+ *    c) Number and street
+ *    d) Locality
+ * 3. Split a single part address into two parts if a comma exists in the first part
+ * 4. Convert a multipart address into two parts by joining all parts except the first with a comma
+ */
+
+const isNotEmpty = compose(not, isEmpty)
+
+const joinAddressFields = addressKeys => compose(join(DELIMITER), filter(isNotEmpty), props(addressKeys))
+
+const constructThoroughfare = joinAddressFields(THOROUGHFARE_PROPS)
+
+const constructBuildingName = joinAddressFields(BUILDING_NAME_PROPS)
+
+const constructLocality = joinAddressFields(LOCALITY_PROPS)
+
+const constructNumberAndStreet = addressFields => `${propOr('', 'BUILDING_NUMBER', addressFields)} ${constructThoroughfare(addressFields)}`.trim()
```

Comment:
This is hard to see in GitHub, is it just doing a string concatenation of the two values (building number and thoroughfare) with a space in-between and then calling trim on the result?

Reply:
Yes that is correct - I am actually not totally happy with this as it is difficult to read .. but I didn't want to introduce more functions as I think that would add further confusion.

I think ideally the two values would be concatenated with a space and wrapped in a trim function.